creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -498,16 +499,25 @@ class SnapbackSM {
     // TODO change to chunked parallel
     await Promise.all(userReplicaSets.map(async (user) => {
       try {
-        const { wallet, endpoint: secondary } = user
+        const { wallet, primary, secondary1, secondary2, endpoint: secondary } = user
 
-        // TODO - throw on null wallet (is this needed?)
+        // Short-circuit if primary is not self - this function is meant to be called from primary to secondaries only
+        if (primary !== this.endpoint) {
+          this.logError(`issueSyncRequests || Can only be called by user's primary. User ${wallet} - replicaset [${primary}, ${secondary1}, ${secondary2}].`)
+          return
+        }
 
-        // Determine if secondary requires a sync by comparing clock values against primary (this node)
-        const userPrimaryClockVal = userPrimaryClockValues[wallet]
-        const userSecondaryClockVal = secondaryNodesToUserClockStatusesMap[secondary][wallet]
-        const syncRequired = !userSecondaryClockVal || (userPrimaryClockVal > userSecondaryClockVal)
+        // Short-circuit on null wallet
+        if (!wallet) {
+          this.logError(`issueSyncRequests || Cannot process null wallet`)
+          return
+        }
 
-        if (syncRequired) {
+        // Determine if secondary requires a sync by comparing clock values against primary (self)
+        const userPrimaryClockVal = userPrimaryClockValues[wallet] || -1
+        const userSecondaryClockVal = secondaryNodesToUserClockStatusesMap[secondary][wallet] || -1
```

Comment:
what if the clock value is 0? I've seen this happen a few times and we might actually want to do an explicit check for null or undefined

Reply:
wow good catch - functionally they are treated the same but i fixed it